bitcoin_client_js/src/lib/appClient.ts

```diff
@@ -26,6 +26,42 @@ enum FrameworkIns {
   CONTINUE_INTERRUPTED = 0x01,
 }
 
+/**
+ * This class represents a partial signature produced by the app during signing.
+ * It always contains the `signature` and the corresponding `pubkey` whose private key
+ * was used for signing; in the case of taproot script paths, it also contains the
+ * tapleaf hash.
+ */
+export class PartialSignature {
+  readonly pubkey: Buffer;
+  readonly signature: Buffer;
+  readonly tapleafHash?: Buffer;
+
+  constructor(pubkey: Buffer, signature: Buffer, tapleafHash?: Buffer) {
+    this.pubkey = pubkey;
+    this.signature = signature;
+    this.tapleafHash = tapleafHash;
```

Comment:
I have no idea of how typescript works, but should you check if `tapleafHash` is actually passed as a parameter before assigning it ?

Reply:
if not passed, it's implicitly `undefined`; and the `?` itself makes its type `Buffer | undefined`, so that's fine.